src/main/kotlin/org/rust/ide/refactoring/move/RsMoveTopLevelItemsDialog.kt

```diff
@@ -196,3 +238,59 @@ class RsMoveItemAndImplsInfo(
 }
 
 val MOVE_TARGET_MOD_KEY: Key<RsMod> = Key("RS_MOVE_TARGET_MOD_KEY")
+val MOVE_TARGET_FILE_PATH_KEY: Key<Path> = Key("RS_MOVE_TARGET_FILE_PATH_KEY")
+
+/** Creates new rust file and attaches it to parent mod */
+private fun createNewRustFile(filePath: Path, project: Project, crateRoot: RsMod?, requestor: Any?): RsFile? {
+    return project.runWriteCommandAction {
+        val fileSystem = (crateRoot as? RsFile)?.virtualFile?.fileSystem ?: LocalFileSystem.getInstance()
+        createNewFile(filePath, fileSystem, requestor) { virtualFile ->
+            val file = virtualFile.toPsiFile(project)?.rustFile ?: return@createNewFile null
+            if (!attachFileToParentMod(file, project, crateRoot)) return@createNewFile null
+            file
+        }
+    }
+}
+
+/** Finds parent mod of [file] and adds mod declaration to it */
+private fun attachFileToParentMod(file: RsFile, project: Project, crateRoot: RsMod?): Boolean {
+    val (parentModOwningDirectory, modName) = if (file.name == RsConstants.MOD_RS_FILE) {
```

Comment:
Super drive-by comment: does this work for both 2015 and 2018 editions?

Reply:
Looks like the only relevant edition difference is that in 2018 edition [module with submodules doesn't have to be named `foo/mod.rs`](https://doc.rust-lang.org/edition-guide/rust-2018/module-system/path-clarity.html#no-more-modrs), so yes, it should work for both editions

Btw Move refactoring itself currently supports only 2018 edition